res/schema.xml

```diff
@@ -446,4 +446,37 @@ METADATA
       );
     </sql>
   </revision>
+  <revision version="29" min_compatible="3">
+    <description>
+      Add tables to support nested crates.
+    </description>
+    <sql>
+      CREATE TABLE IF NOT EXISTS crateClosure (
+        parentId INTEGER,
+        childId INTEGER,
+        depth INTEGER
+      );
+
+      CREATE TABLE IF NOT EXISTS cratePath (
+        crateId INTEGER,
+        idPath TEXT,
+        namePath TEXT
+      );
+
+      ALTER TABLE crates RENAME TO old_crates;
```

Comment:
apparently you can't alter a column's unique status, so I had to create the crates table again this time with the name not unique